src/sql/src/names.rs

```diff
@@ -519,6 +520,35 @@ impl std::fmt::Display for ResolvedItemName {
     }
 }
 
+#[derive(Debug, Clone, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ResolvedColumnName {
```

Comment:
OOC should this be added to `ResolvedObjectName` here https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/names.rs#L780 as well?

Reply:
nvm ^, since it's not an object and will not have a corresponding `ObjectId`

Reply:
Good thought! I noodled on this for a bit and we could maybe add this to `ResolvedObjectName` and return the relation's ID, there didn't seem to be a need to do this at the moment, so I held off